docling: init at 2.17.0

happysalada opened the pull request:
## Things done

opening this early just in case anyone is available to help with anything.
This is still missing
- deepsearch-glm
- pypdfium2
so putting this in draftmode for now.
@drupol you might be interested in this.

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

drupol commented on pkgs/by-name/do/docling/package.nix:
```diff
@@ -0,0 +1,75 @@
+  src = fetchFromGitHub {
+    owner = "DS4SD";
+    repo = "docling";
+    rev = "v${version}";
```
We can now use `tag`

drupol commented on pkgs/by-name/do/docling/package.nix:
```diff
@@ -0,0 +1,75 @@
+  pythonImportsCheck = [
+    "docling"
+  ];
+
```
No tests ? If not, maybe it would be a good idea to add doChecks to false with a comment.

happysalada replied:
I was waiting for this to build before doing the tests.
Because of tje two missing dependencies this wont work.
I checked yesterday, and it looks pdfium2 is packaged as part of a PR that was never merged, so we should be able to copy tyat.
However the other dependency is pulling different repos to compile them with cmake, its going to be a pain to actually package.
But once these two dependencies are in, yes, i should try to enable tje tests.

booxter replied:
I suspect some of you folks may want to check https://github.com/NixOS/nixpkgs/pull/374461 where I package pypdfium2. I need it for docling (which is needed for instructlab) too but decided not to boil an ocean in a single PR and post in pieces.

booxter commented:
Both pypdfium2 and deepsearch-glm are packaged. Please include them. We'll also have to enable the test suite.

happysalada committed:
python3Modules.docling-core: 2.3.2 -> 2.16.1

happysalada committed:
python3Packages.docling-parse: 2.0.3 -> 3.1.2

happysalada committed:
python3Packages.docling-ibm-models: 2.0.4 -> 3.3.0

docling-ibm-models: fix read onlyy file system

docling-ibm-models: disable broken tests

happysalada commented:
@booxter thanks for the ping I noticed by coincidence that this was done this morning, great work!

@drupol I struggled in an update to one package because of a missing semchunk. Ideally I would like to leave one of the tests disabled so we can merge the PR and try to figure out how to package semchunk and re-enable the tests in a separate PR.

I've successfully built and test a --help on x86_64 darwin.

drupol commented:
Sure let's go ahead.

booxter commented:
FYI the Darwin build issue with pypdfium2 is to be fixed in https://github.com/NixOS/nixpkgs/pull/378177

happysalada commented:
ok, I've merged the pypdfium2 fix and rebased on top of it so my commit to it disappears.

I've added the tests to docling and made sure they work.
I've made a small fix to docling-ibm-models tests

I've built and tested again on x86_64 darwin

happysalada commented:
The linux failures were due to some tests in ibm-models requiring network access apparently. Disabled those.

happysalada commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 359783`

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 18 packages built:</summary>
  <ul>
    <li>docling</li>
    <li>docling.dist</li>
    <li>python312Packages.deepsearch-toolkit</li>
    <li>python312Packages.deepsearch-toolkit.dist</li>
    <li>python312Packages.docling-core</li>
    <li>python312Packages.docling-core.dist</li>
    <li>python312Packages.docling-ibm-models</li>
    <li>python312Packages.docling-ibm-models.dist</li>
    <li>python312Packages.docling-parse</li>
    <li>python312Packages.docling-parse.dist</li>
    <li>python313Packages.deepsearch-toolkit</li>
    <li>python313Packages.deepsearch-toolkit.dist</li>
    <li>python313Packages.docling-core</li>
    <li>python313Packages.docling-core.dist</li>
    <li>python313Packages.docling-ibm-models</li>
    <li>python313Packages.docling-ibm-models.dist</li>
    <li>python313Packages.docling-parse</li>
    <li>python313Packages.docling-parse.dist</li>
  </ul>
</details>

happysalada commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 359783`

---
### `x86_64-linux`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>docling</li>
    <li>docling.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 16 packages built:</summary>
  <ul>
    <li>python312Packages.deepsearch-toolkit</li>
    <li>python312Packages.deepsearch-toolkit.dist</li>
    <li>python312Packages.docling-core</li>
    <li>python312Packages.docling-core.dist</li>
    <li>python312Packages.docling-ibm-models</li>
    <li>python312Packages.docling-ibm-models.dist</li>
    <li>python312Packages.docling-parse</li>
    <li>python312Packages.docling-parse.dist</li>
    <li>python313Packages.deepsearch-toolkit</li>
    <li>python313Packages.deepsearch-toolkit.dist</li>
    <li>python313Packages.docling-core</li>
    <li>python313Packages.docling-core.dist</li>
    <li>python313Packages.docling-ibm-models</li>
    <li>python313Packages.docling-ibm-models.dist</li>
    <li>python313Packages.docling-parse</li>
    <li>python313Packages.docling-parse.dist</li>
  </ul>
</details>

happysalada commented:
Failure due to similar tests failing only on linux, this review should be the last

happysalada commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 359783`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 18 packages built:</summary>
  <ul>
    <li>docling</li>
    <li>docling.dist</li>
    <li>python312Packages.deepsearch-toolkit</li>
    <li>python312Packages.deepsearch-toolkit.dist</li>
    <li>python312Packages.docling-core</li>
    <li>python312Packages.docling-core.dist</li>
    <li>python312Packages.docling-ibm-models</li>
    <li>python312Packages.docling-ibm-models.dist</li>
    <li>python312Packages.docling-parse</li>
    <li>python312Packages.docling-parse.dist</li>
    <li>python313Packages.deepsearch-toolkit</li>
    <li>python313Packages.deepsearch-toolkit.dist</li>
    <li>python313Packages.docling-core</li>
    <li>python313Packages.docling-core.dist</li>
    <li>python313Packages.docling-ibm-models</li>
    <li>python313Packages.docling-ibm-models.dist</li>
    <li>python313Packages.docling-parse</li>
    <li>python313Packages.docling-parse.dist</li>
  </ul>
</details>

happysalada commented:
@booxter thanks again for doing the hard work of packaging deepsearch-glm and pypdfium2

booxter commented:
I am going to review this with some more attention, but the first thing that pops into my view is that `docling` here is defined as (just) an application, while it should be both a module to import and an application. You can put.

```
docling = toPythonModule (pkgs.docling.override {
    python3 = python;
  });
```

into `pkgs/top-level/python-packages.nix` to make it work as both, I think.

happysalada commented:
The other way around makes more sense i think to have the module in the module and then have a topythonapplication

I didnt know it was a module as well, noted !

happysalada committed:
docling: init at 2.17.0

happysalada commented:
I've added a python module and built the application from the module.
Let me know if anything else.
I don't think there was very much done in this PR, it's pretty much just tedious work, updating some modules and then adding a new one.

happysalada commented:
I dont think there is a lot done in this PR, would you mind if i merged in the next day or so ?

happysalada commented:
Opening a follow up PR to update docling, that I will leave open until the weekend most likely. If there is any feedback that you see on this PR that you feel is important, feel free to comment here and I will fix it into the follow up PR.
https://github.com/NixOS/nixpkgs/pull/379596

booxter commented on pkgs/development/python-modules/docling-core/default.nix:
```diff
@@ -47,6 +59,8 @@ buildPythonPackage rec {
     "docling_core"
   ];
 
+  doCheck = false;
```
Any particular reason why tests were disabled here?

booxter replied:
https://github.com/NixOS/nixpkgs/pull/379628
